packages/protocol/contracts/psp/AirnodeSponsorPspAllocator.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "../rrp/interfaces/IAirnodeRrp.sol";
+import "./AirnodePsp.sol";
+
+// It would be nice if slots could also expire, similar to whitelisting expiration
+contract AirnodeSponsorPspAllocator {
+    struct Slot {
+        bytes32 subscriptionId;
+        address sponsor;
+    }
+
+    IAirnodeRrp public airnodeRrp;
+    AirnodePsp public airnodePsp;
+    mapping(address => mapping(address => uint256))
+        public airnodeToSponsorToNoOfTotalSlots;
+    mapping(address => mapping(address => uint256))
+        public airnodeToSponsorToNoOfUsedSlots;
```

Comment:
I just posted a message in the DaoPspAllocator about having something like this in there because I reviewed it before this one 🤭